### PR TITLE
Some suggestions for Sam's rekeying fixes

### PR DESCRIPTION
**Pull request by diegoreymendez**
This PR was created to suggest review changes to an open PR.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift [hunk 1]
```diff
@@ -134,6 +134,7 @@ final class NetworkProtectionConnectionTester {
     func stop() async {
         os_log("🔴 Stopping connection tester", log: log)
         await stopScheduledTimer()
+        isRunning = false
```
Comment (diegoreymendez) [inline Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift:137]: This should not be about the internal timer.  It should instead be a flag to signal the monitor in its entirety is running.

For this reason, this flag should be set to `false` only on the public `stop()` interface.

File: Sources/NetworkProtection/PacketTunnelProvider.swift [hunk 2]
```diff
@@ -150,7 +152,10 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
     }
 
     private func rekeyIfExpired() async {
+        os_log("Checking if rekey is necessary...", log: .networkProtectionKeyManagement)
```
Comment (samsymons) [inline Sources/NetworkProtection/PacketTunnelProvider.swift:155]: Glad to see these tbh, I added them myself locally when I was debugging this but discarded the changes before review.

File: Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift [hunk 3]
```diff
@@ -197,12 +198,10 @@ final class NetworkProtectionConnectionTester {
             self?.timer = nil
         }
 
-        isRunning = true
```
Comment (diegoreymendez) [inline Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift:200]: This was a really good catch in your PR, and it took me a while to understand why it was needed.  Good catch!

File: Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift [hunk 4]
```diff
@@ -197,12 +198,10 @@ final class NetworkProtectionConnectionTester {
...
         timer.resume()
     }
 
     private func stopScheduledTimer() async {
-        isRunning = false
```
Comment (diegoreymendez) [inline Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift:205]: This is only called internally and should not signal that the monitor isn't running.

File: Sources/NetworkProtection/PacketTunnelProvider.swift [hunk 5]
```diff
@@ -756,7 +762,7 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
 
         switch message {
         case .request(let request):
-            handleRequest(request)
+            handleRequest(request, completionHandler: completionHandler)
```
Comment (diegoreymendez) [inline Sources/NetworkProtection/PacketTunnelProvider.swift:765]: Ouch.

File: Sources/NetworkProtection/PacketTunnelProvider.swift [hunk 6]
```diff
@@ -862,6 +868,7 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                 .setNetworkPathChange,
                 .setDisableRekeying:
             // Intentional no-op, as some setting changes don't require any further operation
+            completionHandler?(nil)
```
Comment (diegoreymendez) [inline Sources/NetworkProtection/PacketTunnelProvider.swift:871]: ...and ouch!

File: Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift (hunk 4 above)
Comment (samsymons) [reply Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift:205]: :+1:

File: Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift (hunk 1 above)
Comment (samsymons) [reply Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift:137]: This makes sense, nice one.

**Review by samsymons — APPROVED** [review]

**Pull request merged**